ocs_ci/utility/kms.py

```diff
@@ -761,36 +764,304 @@ def create_vault_csi_kms_token(
+            buf = json.loads(conn_str)
+            buf["vaultAddress"] = f"https://{self.vault_server}:{self.port}"
+            buf["vaultBackendPath"] = self.vault_backend_path
+            buf["vaultCAFromsecret"] = get_default_if_keyval_empty(
+                config.ENV_DATA, "VAULT_CACERT", defaults.VAULT_DEFAULT_CA_CERT
+            )
+            buf["vaultClientCertFromSecret"] = self.client_cert_name
+            buf["vaultClientCertKeyFromSecret"] = self.client_key_name
+            if self.vault_namespace:
+                buf["vaultNamespace"] = self.vault_namespace
+            if self.vault_kube_auth_path:
+                buf["vaultAuthPath"] = self.vault_kube_auth_path
+            else:
+                buf.pop("vaultAuthPath")
+            if self.vault_kube_auth_namespace:
+                buf["vaultAuthNamespace"] = self.vault_kube_auth_namespace
+            else:
+                buf.pop("vaultAuthNamespace")
+            csi_kms_conn_details["data"]["vault-tenant-sa"] = json.dumps(buf)
 
-        csi_kms_conn_details["data"]["1-vault"] = json.dumps(buf)
         csi_kms_conn_details["metadata"]["namespace"] = namespace
         self.create_resource(csi_kms_conn_details, prefix="csikmsconn")
 
+    def create_token_reviewer_resources(self):
+        """
+        This function will create the rbd-csi-vault-token-review SA, clusterRole and
+        clusterRoleBindings required for the kubernetes auth method with vaulttenantsa
+        encryption type.
+
+        """
+
+        try:
+            rbd_vault_token_reviewer = templating.load_yaml(
+                constants.RBD_CSI_VAULT_TOKEN_REVIEWER, multi_document=True
+            )
+            self.create_resource(rbd_vault_token_reviewer, prefix="rbd-token-review")
+            logger.info("rbd-csi-vault-token-reviewer resources created successfully")
+
+        except CommandFailed as cfe:
+            if "AlreadyExists" in str(cfe):
+                logger.info("rbd-csi-vault-token-reviewer resources already exists")
+            else:
+                raise
+
+    def create_tenant_sa(self, namespace):
+        """
+        This function will create the serviceaccount in the tenant namespace to
+        authenticate to Vault when vaulttenantsa KMS type is used for PV encryption.
+
+        Args:
+            namespace (str): The tenant namespace where the service account will be created
+        """
+
+        tenant_sa = templating.load_yaml(constants.RBD_CSI_VAULT_TENANT_SA)
+        tenant_sa["metadata"]["namespace"] = namespace
+        self.create_resource(tenant_sa, prefix="tenant-sa")
+        logger.info("Tenant SA ceph-csi-vault-sa created successfully")
+
+    def create_tenant_configmap(
+        self,
+        tenant_namespace,
+        vault_backend=None,
+        vault_backend_path=None,
+        vault_namespace=None,
+        vault_role=None,
+        vault_auth_path=None,
+        vault_auth_namespace=None,
+    ):
```

Comment:
I think you can reduce number of lines in this function.
```suggestion

from ocs_ci.framework import  merge_dict


   def create_tenant_configmap(
       self,
       tenant_namespace,
       **vault_config
    )
        tenant_cm = templating.load_yaml(
            constants.RBD_CSI_VAULT_TENANT_CONFIGMAP
        )
        tenant_cm["metadata"]["namespace"] = tenant_namespace
        
        merge_dict(tenant_cm, vault_config)
        for k in tenant_cm.copy():
            if not tenant_cm[k]:
                tenant_cm.pop(k)
```

Reply:
Done